accepted/00076-configuration-channels-visibility.md

```diff
@@ -0,0 +1,201 @@
+- Feature Name: Enhance visibility of connected Configuration State Channels
+- Start Date: 2020-12-17
+- RFC PR:
+
+# Summary
+[summary]: #summary
+
+Provide visibility of **inherited** Configuration State Channels connections at a System Details level. The reverse information as well should be provided by giving visibility of those systems they are connected (directly or by **inheritance**) to a certain Configuration State Channel at a Channel Details level.
+
+*Note: everything discussed in this RFC applies to Salt clients only. Traditional clients do not inherit Configuration State Channels.*
+
+# Motivation
+[motivation]: #motivation
+
+## Why are we doing this?
+
+By definition a Configuration State Channel is a pool of configuration salt states resources. Once a channel of these types is created, it is possible to subscribe systems to it in order to deploy the content of that channel to them. Alternatively, channels can also be assigned to System Groups and/or to Organizations.
+
+Collecting the possibilities as described above, we can see that:
+- a channel can be assigned directly to a system `channel --> system`
+- a channel can be assigned to a system group `channel --> group`
+- a channel can be assigned to an organization `channel --> org`
+
+On top of this it has to be considered that by design a System is part of an Organization `system --> org`, and it can also be part of a System Group `system --> group`.
+
+The final channel assignements result, from a system perspective, can be:
+1. `system --> channel`
+2. `system --> group --> channel`
+3. `system --> org --> channel`
+
+That said, in the current implementation there is no way to figure out which channels are assigned to a system other than the directly assigned ones (1.).
+
+And here it is the problem this RFC is trying to describe to solve: apart from the direct assignment, the System inherits channels (thus the content) from the Organization it is part of, and the same goes for the System Groups he is part of, but this inheritance is not visible in the Web UI nor in the XML RPC API.
+
+Note: in addition to what has been described so far, for what concerns the Configuration State Channels case, behind all the Salt States collected from the channels, there are also a bunch of default Salt States, the ones the server has to supply during the registration of a salt minion client. Those are not *assigned* in any way because they are not part of any Configuration State Channel, they do just exists due to the logic implementation of a client registration using Salt. In the end, those are not visible Salt States anyway, and the System will be affected by them during the registration, so from the System perspective an additional source of Salt States could be defined as directly assigned: a non-manageable source behaving as a Configuration State Channel that will be named *registration-states*.
```

Comment:
Initially I left part of your point in the _unresolved questions_ section, as I don't yet see a *strong* need to represent these states for two reasons:
1- we don't know in advance what can be part of such a bunch of states
2- we would need to just represent statically a generic `internal management states` entity that would be valid for all the registered system, ever.

Therefore, if we agree on this, I'll adapt the RFC removing the _unresolved question_ point and adding to the design the fact that this information will be statically presented in a Note in `Systems > Details > States` page: that is the relevant place where it makes sense to remember _Hey, keep in mind this systems has also a couple of internal states due to internal logic management, ..._ . By doing so we also keep it open for the future the possibility to implement from there a detailed view for those internal states only, a view we don't have and we don't need at the moment.

Bottom line, I fully agree it makes much more sense a generic `internal states` instead of `registration states`. Thanks for your point :+1:

Reply:
Think of the future Ansible integration. Users managing their systems mainly with Ansible, where Salt will be only used for those `internal states`, definitely want to know what those states are to avoid the Ansible playbooks and Uyuni internal states conflicting.